src/IO/H5/File.hpp

```diff
@@ -392,7 +392,16 @@ const ObjectType& H5File<Access_t>::convert_to_derived(
 template <AccessType Access_t>
 template <typename ObjectType>
 std::tuple<bool, detail::OpenGroup, std::string>
-H5File<Access_t>::check_if_object_exists(const std::string& path) const {
+H5File<Access_t>::check_if_object_exists(std::string path) const {
+  // Normalize path to have a leading slash and the correct extension
+  if (path.front() != '/') {
+    path = '/' + path;
```

Comment:
Can you add a test of this normalization in the c++ tests?

Reply:
done and squashed in